src/components/DateRangePicker.jsx

```diff
@@ -481,6 +481,7 @@ class DateRangePicker extends React.PureComponent {
     /* eslint-disable jsx-a11y/click-events-have-key-events */
     return (
       <div
+        key="day-picker"
```

Comment:
similarly, this is never rendered into an array, so the key here has no effect. why is this here?

Reply:
Keys can let us control how React create new instances of components. This is most commonly used for preventing unnecessary component mount/unmounts when rendering arrays of components, but is not _only_ for arrays. 

Ken Dodds has a clear example of why a key might be useful in non-array components to force a new instance [here](https://kentcdodds.com/blog/understanding-reacts-key-prop).

In this case I am doing the precise opposite: by specifying a key I give React the information it needs to not create a new instance even though I am doing something which means it normally would (move a React element to a new position in the tree). If you check out my branch and play around with it in storybook with and without either `key` prop you'll see that it flickers without the `key` prop on reverse-tabbing (caused by un/re-mount). If it isn't apparent you can add logging to the existing `ref` callback and see the absence of a `key` prop causes additional DOM changes.

Reply:
This is a single hardcoded string though - what happens if I have multiple DateRangePicker instances on the page?

Reply:
There's no issue here across multiple instances. Keys only have to be unique across siblings within a given element but not globally either across elements or instances of those elements. They key is applied to an element which

Just to be sure I tested this with multiple instances on the same page and they behave completely independently as expected. Do you have a specific concern for unwanted behavior that we can check for?

Here is what I am trying to achieve with the key:
```tsx
// Before tab
<DateInput ... />
<DatePicker ... />
<DateInput ... />

// After tab
<DateInput ... />
<DateInput ... />
<DatePicker ... />
```

Without a key on `DatePicker` React is going to create a new instance which causes a flicker. With the key React will not create a new instance and there won't be a flicker (aside: during forward navigation the blur handler will still cause a flicker, which is a separate issue. Back-tab-navigation is the better demonstration here).

Reply:
The element with a key is within the implementation of this component. So while it would be possible for a future editor of this library to encounter this if they rendered two drop-downs (albeit unlikely since a single DateRangePicker shouldn't want multiple drop-downs) it is not possible for a user of the library.